packages/block-library/src/quote/edit.js

```diff
@@ -46,67 +61,53 @@ export default function QuoteEdit( {
 						setAttributes( { align: nextAlign } );
 					} }
 				/>
-			</BlockControls>
-			<BlockQuotation { ...blockProps }>
-				<RichText
-					identifier="value"
-					multiline
-					value={ value }
-					onChange={ ( nextValue ) =>
-						setAttributes( {
-							value: nextValue,
-						} )
-					}
-					onMerge={ mergeBlocks }
-					onRemove={ ( forward ) => {
-						const hasEmptyCitation =
-							! citation || citation.length === 0;
-						if ( ! forward && hasEmptyCitation ) {
-							onReplace( [] );
+				<ToolbarGroup>
+					<ToolbarButton
+						isActive={ hasAttribution }
+						label={ __( 'Toggle attribution visibility' ) }
+						onClick={ () =>
+							setAttributes( {
+								attribution: hasAttribution ? null : '',
+							} )
 						}
-					} }
-					aria-label={ __( 'Quote text' ) }
-					placeholder={
-						// translators: placeholder text used for the quote
-						__( 'Add quote' )
-					}
-					onReplace={ onReplace }
-					onSplit={ ( piece ) =>
-						createBlock( 'core/quote', {
-							...attributes,
-							value: piece,
-						} )
-					}
-					__unstableOnSplitMiddle={ () =>
-						createBlock( 'core/paragraph' )
-					}
-					textAlign={ align }
-				/>
-				{ ( ! RichText.isEmpty( citation ) || isSelected ) && (
+					>
+						{ __( 'Add attribution' ) }
+					</ToolbarButton>
+				</ToolbarGroup>
+			</BlockControls>
+			{ showAttribution ? (
+				<figure { ...innerBlocksProps }>
+					<BlockQuotation>
+						{ innerBlocksProps.children }
```

Comment:
This is not how `innerBlocksProps` should be used. `innerBlocksProps.children` must be direct children to whatever element `innerBlocksProps` is spread on. Both for the edit and save functions.

We should probably log an error if that doesn't happen.

Reply:
This is news to me. There should definitely be a warning about this in the docs for this hook.

So I'm guessing `innerBlocksProps` should be spread on the `BlockQuotation` instead?